### PR TITLE
Fixes json serialization issues

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes json serialization so it can handle numpy arrays.

Closes: #1399
Closes: #1408

### CODE REVIEW

File: gradio/components.py [hunk 1]
```diff
@@ -2676,13 +2677,25 @@ def __init__(
...
         """
-        self.value = value
+        self.value = orjson.loads(value) if isinstance(value, str) else value
+        try:
+            orjson.dumps(self.value, option=orjson.OPT_SERIALIZE_NUMPY)
+        except TypeError:
+            raise ValueError(
+                f"Cannot set initial value because type: {type(value)} is"
+                "not JSON serializable. Instead, you can set the initial "
+                "value to be None, and in your Python function, change it"
```
Comment (Kludex) [inline gradio/components.py:2687]:
```suggestion
                "value to be None, and in your Python function, change it "
```

**Review by Kludex — COMMENTED** [review]
👀

**Comment by aliabid94** [comment]
orjson was not added to requirements.txt. Do we need to add a new library? Is it not sufficient to test if a value is a numpy.*, and use .tolist() or .item() to get the python native equivalent?

**Comment by FarukOzderim** [comment]
We've been already using `orjson `in routes.py and there were no problems about it.

> orjson was not added to requirements.txt. Do we need to add a new library? Is it not sufficient to test if a value is a numpy.*, and use .tolist() or .item() to get the python native equivalent?

File: gradio/components.py [hunk 2]
```diff
@@ -2676,13 +2677,25 @@ def __init__(
         Parameters:
         value (Any): the initial value of the state.
         """
-        self.value = value
+        self.value = orjson.loads(value) if isinstance(value, str) else value
+        try:
+            orjson.dumps(self.value, option=orjson.OPT_SERIALIZE_NUMPY)
+        except TypeError:
+            raise ValueError(
+                f"Cannot set initial value because type: {type(value)} is"
```
Comment (abidlabs) [inline gradio/components.py:2685]:
```suggestion
                f"Cannot set initial value because type: {type(value)} is "
```

**Pull request merged**